core/src/test/java/kafka/test/api/ShareConsumerTest.java

```diff
@@ -344,6 +349,7 @@ private void testHeadersSerializeDeserialize(Serializer<byte[]> serializer, Dese
 
         List<ConsumerRecord<byte[], byte[]>> records = consumeRecords(shareConsumer, numRecords);
         assertEquals(numRecords, records.size());
+        producer.close();
```

Comment:
And again. Please check all instances of share consumer are closed.

Reply:
Okay, I have updated the code and pushed again now.